pyccel/codegen/printing/ccode.py

```diff
@@ -1033,15 +1036,16 @@ def _print_Constant(self, expr):
         return self._print(val)
 
     def _print_Return(self, expr):
-        code = ''
         args = [VariableAddress(a) if self.stored_in_c_pointer(a) else a for a in expr.expr]
-        if expr.stmt:
-            code += self._print(expr.stmt)+'\n'
+        if len(args) > 1:
+            if expr.stmt:
+                return self._print(expr.stmt)+'\n'+'return 0;'
+            return 'return 0;'
         if len(args) == 1:
-            code +='return {0};'.format(self._print(args[0]))
-        elif len(args) > 1:
-            code += 'return 0;'
-        return code
+            if expr.stmt:
+                return 'return {0};'.format(self._print(expr.stmt.body[0].rhs))
```

Comment:
i think we should handle both the codeBlock and the Assign

Reply:
In that case your code won't work if `expr.stmt` is an `Assign`. Another option would be for the `expr.stmt` to always be a `CodeBlock`. Is it ever an `Assign` currently?

Reply:
thats what i observe its always `CodeBlock`, i just need test cases when `expr.stmt` comes as an Assign.

Reply:
should i handle it as in Python printer?

Reply:
If it is currently always a `CodeBlock` then I think it makes sense for this to be the expected behaviour. So I suggest that we update `Return.__new__` to only accept a `CodeBlock` object, or to put an `Assign` argument into a `CodeBlock`

Reply:
>should i handle it as in Python printer?

I'm not sure I see what you mean here. In the python printer you return multiple objects which is obviously not possible in c

Reply:
Ah I have just seen what you mean. This is also possible but I think it would be easier to just ensure that `Return.stmt` always contains a `CodeBlock`, that way we don't have to maintain 2 versions everywhere despite the fact that only 1 is used

Reply:
If we are keeping `CodeBlock` though, it is important that you collect the rhs from the **last** statement not the first. It is possible that more complicated expressions may be converted to multiple lines. E.g:
```
return [i for i in range(4)]
```
would probably be translated to something like:
```
Allocate(Dummy_0001, 4)
For(Dummy_0002, range(4), Assign(Dummy_001[Dummy_0002], Dummy_0002))
Assign(Out_0001, Dummy_0001)
```
(This is probably not an accurate translation but I can't think of a better example right now) You can see that in a case like this it is not the first line of the block which interests you

Reply:
yes it will be the last statement